### PR TITLE
add decimal.js@7.1.0 w/ npm auto-update

### LINKED ISSUE #9931: [Request] Add decimal.js
**Library name:** decimal.js
**Git repository url:** https://github.com/MikeMcl/decimal.js
**npm package url(optional):** https://www.npmjs.com/package/decimal.js
**License(s):** MIT
**Official homepage:** http://mikemcl.github.io/decimal.js/

### PR DESCRIPTION
**Pull request by qwe19272375**
close #9931, cc @MikeMcl

Pull request for issue: #9931 
Related issue(s): 

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/MikeMcl/decimal.js
 * Official website (optional, not the repository): https://mikemcl.github.io/decimal.js/
 * NPM package url (optional): https://www.npmjs.com/package/decimal.js
 * License and its reference: MIT https://github.com/MikeMcl/decimal.js#licence
 * GitHub / Bitbucket popularity (required): 
   - Count of stars: 1072
   - Count of watchers: 28
   - Count of forks: 72
 * NPM download stats (optional):
   - Downloads in the last day: 3477
   - Downloads in the last week: 19317
   - Downloads in the last month: 92453

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by qwe19272375** [comment]
@nolsherry Please help me review this PR, thanks!

**Comment by qwe19272375** [comment]
`doc/decimal.js.map` file added and `package.json` modified. @nolsherry Please help me review it, thanks!

**Comment by nolsherry** [comment]
LGTM, piing @cdnjs/intern2.

**Comment by qwe19272375** [comment]
Modify filemap using two basepath. @nolsherry Please help me review it, thanks!

**Comment by nolsherry** [comment]
LGTM, ping @cdnjs/intern2.

**Comment by kennynaoh** [comment]
LGTM

File: ajax/libs/decimal.js/package.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  "npmName": "decimal.js",
+  "npmFileMap": [
+    {
+      "basePath": "",
+      "files": [
+        "decimal*"
+      ]
+    },
+    {
+      "basePath": "doc",
```
Comment (PeterDaveHello) [inline ajax/libs/decimal.js/package.json]: This is totally wrong. @kennynaoh you are going to break the structure.

File: ajax/libs/decimal.js/7.1.0/decimal.min.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
+//# sourceMappingURL=doc/decimal.js.map
```
Comment (PeterDaveHello) [inline ajax/libs/decimal.js/7.1.0/decimal.min.js]: @kennynaoh take a look at this comment, why should we use `doc` as base path?

**Pull request merged**